A28-xds-traffic-splitting-and-routing.md

```diff
@@ -0,0 +1,668 @@
+        *   The presence of this field will cause a NACK
+*   Can have
+    [runtime\_fraction](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L428)
+    *   gRPC doesn’t support runtime config, so only
+        [default\_value](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/core/base.proto#L369)
+        will be used,
+        [runtime\_key](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/core/base.proto#L372)
+        will be ignored
+*   Can have
+    [query\_parameters](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L442)
+    *   Note: the entire route containing this will be ignored, regardless of
+        the other matchers, this is the same as this always evaluates to false,
+        because gRPC does not and will not support query parameters
+*   Can have
+    [grpc](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L447)
+    *   This matcher will be ignored (but the other matchers in the route will
+        be evaluated)
+*   Can have
+    [tls_context](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L453)
+    *   This matcher will be ignored (but the other matchers in the route will
+        be evalulated). We may add support for this feature later, but the
+        service owners need to be careful how they start using it, because the
+        older clients will ignore it
+*   Must not have
+    [case\_sensitive](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L410)
+    set to false (OK to be unset, default value is true)
+    *   Setting this to false will cause a NACK
+
+The route’s [action
+field](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L181)
+must be
+[route](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L185),
+and its
+[cluster\_specifier](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L696):
+*   Can be
+    [Cluster](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L701)
+*   Can be
+    [Weighted\_clusters](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L719)
+    *   The sum of
+        [weights](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L278)
+        must add up to the
+        [total\_weight](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L335).
+*   Can be
+    [cluster\_header](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L712)
+    *   Note: the route containing this action will be ignored. This feature can
+        be accomplished by header matching.
```

Comment:
@mattklein123 @envoyproxy/api-shepherds @envoyproxy/udpa-wg FYI the above section in this document provides an interesting example of how xDS `RouteConfiguration` is in practice interpreted by another client (in this case gRPC).